src/test/java/org/opentripplanner/routing/algorithm/via/TestViaRoutingWorker.java

```diff
@@ -0,0 +1,170 @@
+package org.opentripplanner.routing.algorithm.via;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.opentripplanner.model.plan.PlanTestConstants.A;
+import static org.opentripplanner.model.plan.PlanTestConstants.B;
+import static org.opentripplanner.model.plan.PlanTestConstants.C;
+import static org.opentripplanner.model.plan.PlanTestConstants.place;
+import static org.opentripplanner.model.plan.TestItineraryBuilder.newItinerary;
+
+import java.time.Duration;
+import java.time.ZonedDateTime;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import org.opentripplanner.model.GenericLocation;
+import org.opentripplanner.model.plan.Itinerary;
+import org.opentripplanner.model.plan.TripPlan;
+import org.opentripplanner.routing.api.request.RouteRequest;
+import org.opentripplanner.routing.api.request.request.JourneyRequest;
+import org.opentripplanner.routing.api.request.request.RouteViaRequest;
+import org.opentripplanner.routing.api.request.request.ViaLocation;
+import org.opentripplanner.routing.api.response.RoutingResponse;
+import org.opentripplanner.util.time.TimeUtils;
+
+/**
+ * Create search from point A to point B via point C. Search will start at 12:00 and will find two
+ * trips (itineraries) from A to C (s1i1, s1i2).
+ * <p>
+ * s1i1 arrives at C 13:00 and s1i2 arrives at C 14:00
+ * <p>
+ * minSlack is 10 min and maxSlack is 45 minutes.
+ * <p>
+ * Search from C to B at 13:15 (first arrival + minSlack) gives 3 results s2i1, s2i2 and s2i3. s2i1
+ * departures at 13:15, s2i2 departures at 13:45 and s2i3 departures at 14:30
+ * <p>
+ * s1i1 should be combined with s2i1 and s2i2. Not s2i3 because maxSlack gives that it shold not
+ * match departures after 13:45 (first arrival + maxSlack).
+ * <p>
+ * s1i2 should be combined only with s2i3 because it arrives after departure on the others.
+ */
+public class TestViaRoutingWorker {
```

Comment:
😻

Reply:
But shouldn't it be called ViaRoutingWorkerTest?